x-pack/plugins/infra/public/pages/metrics/index.tsx

```diff
@@ -87,34 +87,34 @@ export const MetricDetail = withTheme(
           );
         }
         const layouts = layoutCreator(this.props.theme);
-        const breadcrumbs = [{ text: nodeName }];
 
         return (
-          <ColumnarPage>
-            <Header
-              appendSections={<InfrastructureBetaBadgeHeaderSection />}
-              breadcrumbs={breadcrumbs}
-            />
-            <WithMetricsTimeUrlState />
-            <DetailPageContent>
-              <WithOptions>
-                {({ sourceId }) => (
-                  <WithMetricsTime resetOnUnmount>
-                    {({
-                      currentTimeRange,
-                      isAutoReloading,
-                      setRangeTime,
-                      startMetricsAutoReload,
-                      stopMetricsAutoReload,
-                    }) => (
-                      <WithMetadata
-                        layouts={layouts}
-                        sourceId={sourceId}
-                        nodeType={nodeType}
-                        nodeId={nodeName}
-                      >
-                        {({ filteredLayouts }) => {
-                          return (
+          <WithOptions>
+            {({ sourceId }) => (
+              <WithMetricsTime resetOnUnmount>
+                {({
+                  currentTimeRange,
+                  isAutoReloading,
+                  setRangeTime,
+                  startMetricsAutoReload,
+                  stopMetricsAutoReload,
+                }) => (
+                  <WithMetadata
+                    layouts={layouts}
+                    sourceId={sourceId}
+                    nodeType={nodeType}
+                    nodeId={nodeName}
+                  >
+                    {({ displayName, filteredLayouts }) => {
```

Comment:
The loading state for the metadata is not handled, so we have intermediate views showing incomplete breadcrumbs and loading dialogs:

![image](https://user-images.githubusercontent.com/973741/49941800-d470e580-fee3-11e8-86d5-6da926b4bf7d.png)

![image](https://user-images.githubusercontent.com/973741/49941843-ee122d00-fee3-11e8-85b9-6d91f8f0afa6.png)

We could fall back to the `id` until the name is loaded?

Reply:
What if it just said "Loading data" and we don't specify anything?

Reply:
That could work, but what will be shown in the breadcrumbs?